docs/updater.md

```diff
@@ -0,0 +1,102 @@
+* `gvk::destroying_compute_pipeline_event`: This event occurs when an outdated compute pipeline is destroyed.
+* `gvk::destroying_ray_tracing_pipeline_event`: This event occurs when an outdated ray tracing pipeline is destroyed.
+* `gvk::destroying_image_event`: This event occurs when an outdated image is destroyed.
+* `gvk::destroying_image_view_event`: This event occurs when an outdated image view is destroyed.
+* `gvk::files_changed_event`: This event occurs when any of the given files have been modified on the file system.
+
+
+## How to use
+
+The event handling and the automatic updater mechanisms are available via `gvk::updater` class. The access to an updater is restricted to objects of type `gvk::invokee`. This means that each `gvk::invokee` object possesses its own `gvk::updater` instance, functioning in isolation from other updater instances. This updater can be made available by the following call:
+```
+mUpdater.emplace();
+```
+The general schema for using an updater is the following:
+
+```
+mUpdater.on(event... events).[update(...)|invoke(...)][.then_on(event... events).[update(...)|invoke(...)]]*
+```
+
+* `events`: a comma seperated list of event objects. The events are in a logical _or_ relationship, which means that the bound callbacks are invoked if _any_ of the events occur.
+* `on(event... events)`: bind the following up `update` or `invoke` call to a list of events.
+* `update(updatee... updatees)`: automatically updates the given comma seperated list of objects once the bound event occurs. The following resources can be updated:
+  * `avk::graphics_pipeline`
+  * `avk::compute_pipeline`
+  * `avk::ray_tracing_pipeline`
+  * `avk::image`
+  * `avk::image_view`
+* `invoke(handler... handlers)`: invokes the callbacks specified by the comma separated list once the bound event occurs. The following function signatures are allowed:
+  * `std::function<void()>`: For general use.
+  * `std::function<void(const avk::graphics_pipeline&)>`: In combination with `gvk::destroying_graphics_pipeline_event`.
+  * `std::function<void(const avk::compute_pipeline&)>`: In combination with `gvk::destroying_compute_pipeline_event`.
+  * `std::function<void(const avk::ray_tracing_pipeline&)>`: In combination with `gvk::destroying_ray_tracing_pipeline_event`.
+  * `std::function<void(const avk::image&)>`: In combination with `gvk::destroying_image_event`.
+  * `std::function<void(const avk::image_view&)>`: In combination with `gvk::destroying_image_view_event`.
+* `then_on(event... events)`: bind the following up `update` or `invoke` call to a list of events. However, unlike `on(event... events)`, the given events will be evaluated _after_ the updates induced by previous `update` or `invoke` calls have finished execution.
+
+It is important to note that the order in which the events are evaluated is not necessarily determined by the order of the events in each separate list.
+
+### Examples
+
```

Comment:
Some sub-headers would be nice for every example.

Reply:
Furthermore, these Examples should not be confused with the "Examples" below. => Maybe rename this section here to "Code Examples" and the section below to "Example Applications"?!